main/2024/CG-12-17.md

```diff
@@ -27,7 +31,201 @@ No registration is required for VC meetings. The meeting is open to CG members o
+
+TL: makes sense. Personally I’m very interested to use whatever comes out of this, just for informal debugging, it seems generally useful.
+
+EG: Here is my email, please reach out if you are interested in attending meetings on this: evgilber@andrew.cmu.edu
+
+ZB (chat): @evgilber@andrew.cmu.edu , @Benjamin Titzer it might be simpler if you create and share a (google) form to gather interested parties.
+
+EG (chat): For those interested in dynamic instrumentation for Wasm, fill out this form! https://forms.gle/AeFTsHEsdv5kKnX88
+
+####  Presentation of [static memory protection](https://github.com/WebAssembly/memory-control/blob/main/proposals/memory-control/static-protection.md) from the [Memory Control](https://github.com/WebAssembly/memory-control/) proposal
+
+BV: Presenting (TODO: slides)
+
+A simple form of memory protection that we think could fit in with relatively low cost.
+
+AR: question: when we say nullptrs don’t trap, that assumes a choice by the compiler. The C compiler can compile it to point to -1 or something. The value of NULL could be anything
+
+KW (chat): My understanding was that LLVM generally makes sure that calls to null function pointers do trap, by starting to populate the funcref table at 1.
+
+NF (chat): yep, what isn't caught by default is reading/writing data through a null pointer
+
+BV: In principle, yes, a toolchain could work around this. But even then, 0 is probably still going to be the most common value.
+
+Security slide
+
+MF (chat): FYI there's a Stage 2 TC39 proposal for immutable ArrayBuffers, which may help with the JS integration for this feature: https://github.com/tc39/proposal-immutable-arraybuffer. But probably only if the design marked a whole memory as read-only.
+
+DG (chat): IMO this proposal could still work in a reasonable way with this proposal, because you can't mix protections when exposing to JS. This proposal in several sub proposals assume that you would have to decouple the backing store with the JSArrayBuffer, so I could see a future JS API index translate and expose the read/write portion, but expose the read-only section as an immutable-buffer
+
+TL (chat): Do you mean this sub-proposal could still work well in the full memory control proposal?
+
+MF (chat): so back the memory with 2 (or 3?) separate ArrayBuffers?
+
+KW: Is the rationale for making it static really just about performance?
+
+BV: making it possible to implement efficient protection when using bounds checks
+
+KW: That’s a benefit of knowing AOT vs having it in a register?
+
+BV: yes we think so but don’t have numbers.
+
+RH: I think the cost of a satanic lower bound is a subtract and a const vs another load. If you can grow the RO region, it shifts the RW region and that affects bounds check elimination and optimizations. That’s the rationale for not growing the RO region.
```

Comment:
satanic? 😈 

I've fixed up a few typos in the Google Doc, and added a link to my slides, so perhaps it would be worth copy-pasting one more time.

Reply:
I had fixed it incorrectly in the doc; it was supposed to be "static" and clearly it got corrected. Now it's good.